lib/config.service.ts

```diff
@@ -243,4 +243,31 @@ export class ConfigService<
   ): options is ConfigGetOptions {
     return options && options?.infer && Object.keys(options).length === 1;
   }
+
+  /**
+   * Update a configuration value in all possible cases(either custom configuration or process environment variable)
+   * based on property path (you can use dot notation to traverse nested object, e.g. "database.host").
+   * If the cache is enabled will not change anything.
+   * @param propertyPath
+   * @param newValue
+   */
+  set(propertyPath: KeyOf<K>, newValue: string) {
+    if(!this.isCacheEnabled) {
+      const validatedEnvValue = get(
+          this.internalConfig[VALIDATED_ENV_PROPNAME],
+          propertyPath,
+      );
+      if (!isUndefined(validatedEnvValue)) {
+        set(this.internalConfig[VALIDATED_ENV_PROPNAME], propertyPath.toString(), newValue)
+      }
+      if (!isUndefined(get(process.env, propertyPath))) {
+        set(process.env, propertyPath.toString(), newValue)
```

Comment:
I might be wrong but don't think that `@nestjs/config` was designed to touch `process.env`...

Reply:
Can you explain your point? I think it can touch `process.env` file, it works on the configuration in general so it would make sense to give the developer the ability to update the `process.env` / configurations values.

Reply:
To me, `@nestjs/config` is a way to retrieve data from `process.env`

But yeah, dotenv do touches process.env

It just a felling of changing `process.env` like that seems wrong as the service has its internal config